perf/lua/column_insert_module.c

```diff
@@ -0,0 +1,238 @@
+#include <lua.h>
+#include <lauxlib.h>
+#include <module.h>
+#include <msgpuck.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "trivia/config.h"
+#include "trivia/util.h"
+
+#ifdef ENABLE_MEMCS_ENGINE
+# include "arrow/abi.h"
+# define ENABLE_BATCH_INSERT 1
+#endif
+
+static struct {
+	int64_t row_count;
+	int64_t column_count;
+	struct {
+		char *name;
+		const char *type;
+		uint64_t *data;
+	} *columns;
+} dataset;
+
+static int
+insert_serial_lua_func(struct lua_State *L)
+{
+	uint32_t space_id = luaL_checkinteger(L, 1);
+	luaL_checktype(L, 2, LUA_TTABLE);
+	lua_getfield(L, 2, "column_count_total");
+	int total_column_count = luaL_checkinteger(L, -1);
+	lua_pop(L, 1);
+	static char tuple_data[1000 * 1000];
+
+	for (int64_t i = 0; i < dataset.row_count; i++) {
+		char *data_end = tuple_data;
+		data_end = mp_encode_array(data_end, total_column_count);
+		int j;
+		for (j = 0; j < dataset.column_count; j++) {
+			uint64_t val = dataset.columns[j].data[i];
+			data_end = mp_encode_uint(data_end, val);
+		}
+		for (; j < total_column_count; j++)
+			data_end = mp_encode_nil(data_end);
```

Comment:
Just a question, is it known how does the creation of tuples prior to each insertion affect the benchmarking results?

I see creating tuples beforehand could require a whole bunch of memory.

Reply:
I think it is more fair to include the tuple creation time into the total measured time. Anyway, it is negligible. E.g. for 100 columns:

<img width="573" alt="Screenshot 2024-07-08 at 15 11 08" src="https://github.com/tarantool/tarantool/assets/5719536/ca9d8a3b-2aee-4121-8715-edc973599e8a">